client/src/components/Playlist/Playlist.tsx

```diff
@@ -25,19 +26,20 @@ function Playlist() {
       if (playlists) {
         dispatch(setPlaylist(playlists)); // set all playlists from user's playlist data
       }
-      if (playlist.selectedPlaylist === '') {
+      if (selectedPlaylist === '') {
         dispatch(setSelectedPlaylist(items[0].id)); // set first playlist as selected
       }
     };
     getPlaylistData();
-  }, [dispatch, playlist.selectedPlaylist]);
+  }, [dispatch, selectedPlaylist]);
 
   return (
     <PlaylistContainer>
       <ul>
-        {playlist.playlist.map(({ name, id }) => {
-          return <li key={id}>{name}</li>;
-        })}
+        {Object.keys(playlist).length > 0 &&
+          playlist.map(({ name, id }) => {
+            return <li key={id}>{name}</li>;
+          })}
```

Comment:
Let's talk more about this block of code at some point... is it necessary to use `Object.keys` here? Can you instead just do 
```suggestion
        {playlist.length &&
          playlist.map(({ name, id }) => {
            return <li key={id}>{name}</li>;
          })}
```